ingestion-beam/src/main/java/com/mozilla/telemetry/io/Read.java

```diff
@@ -35,15 +35,25 @@
   public static class PubsubInput extends Read {
 
     private final ValueProvider<String> subscription;
+    private final String idAttribute;
 
-    public PubsubInput(ValueProvider<String> subscription) {
+    public PubsubInput(ValueProvider<String> subscription, String idAttribute) {
       this.subscription = subscription;
+      this.idAttribute = idAttribute;
     }
 
     @Override
     public PCollection<PubsubMessage> expand(PBegin input) {
+      final PubsubIO.Read<PubsubMessage> readBase = PubsubIO
+          .readMessagesWithAttributesAndMessageId();
+      final PubsubIO.Read<PubsubMessage> read;
+      if (idAttribute == null) {
+        read = readBase;
+      } else {
+        read = readBase.withIdAttribute(idAttribute);
+      }
```

Comment:
`getIdAttribute` is marked as `@Nullable` in PubsubIO, which means `idAttribute` is allowed to be null. I think this means it's also possible to pass in `null` to `withIdAttribute`, so this logic is hopefully unnecessary.

Reply:
yep, works as expected when null.